src/main/scala/jtag/Utils.scala

```diff
@@ -12,40 +13,23 @@ class Tristate extends Bundle {
   val driven = Bool()  // active high, pin is hi-Z when driven is low
 }
 
-class NegativeEdgeLatch[T <: Data](clock: Clock, dataType: T)
-    extends Module(override_clock=Some(clock)) {
-  class IoClass extends Bundle {
-    val next = Input(dataType)
-    val enable = Input(Bool())
-    val output = Output(dataType)
-  }
-  val io = IO(new IoClass)
-
-  val reg = Reg(dataType)
-  when (io.enable) {
-    reg := io.next
-  }
-  io.output := reg
-}
-
 /** Generates a register that updates on the falling edge of the input clock signal.
   */
 object NegativeEdgeLatch {
   def apply[T <: Data](clock: Clock, next: T, enable: Bool=true.B, name: Option[String] = None): T = {
-    // TODO better init passing once in-module multiclock support improves
-    val latch_module = Module(new NegativeEdgeLatch((!clock.asUInt).asClock, next.cloneType))
-    name.foreach(latch_module.suggestName(_))
-    latch_module.io.next := next
-    latch_module.io.enable := enable
-    latch_module.io.output
+    // TODO pass in initial value as well
+    withClock((!clock.asUInt).asClock) {
+      val reg = RegEnable(next = next, enable = enable)
+      name.foreach{reg.suggestName(_)}
+      reg
+    }
   }
 }
 
 /** A module that counts transitions on the input clock line, used as a basic sanity check and
   * debug indicator clock-crossing designs.
   */
-class ClockedCounter(modClock: Clock, counts: BigInt, init: Option[BigInt])
-    extends Module(override_clock=Some(modClock)) {
+class ClockedCounter(counts: BigInt, init: Option[BigInt]) extends Module {
```

Comment:
Now that we're using withClock, is this any different than using Counter(true.B, counts) from Chisel?

Reply:
Oh, I guess the ability to set the reset value to something nonzero is a difference.